### PR TITLE
Improve FakeTimeProvider documentation, remove redundant tests

### PR DESCRIPTION
**Pull request by amadeuszl**
Improve `FakeTimeProvider` documentation, remove redundant tests, refactor tests methods.

I updated test method names as well to make them more readable, so that the code can also serve as nice documentation and reference to the users.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/5683)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md [hunk 1]
```diff
@@ -33,31 +33,109 @@ public void Advance(TimeSpan delta)
...
+        timeProvider.SetUtcNow(timeProvider.GetUtcNow() + TimeSpan.FromSeconds(2));
+
+        // The item should now be expired
+        found = cache.TryGetValue("key1", out value);
+        Assert.False(found);
+    }
+}
 ```
 
 ## SynchronizationContext in xUnit Tests
```
Comment (eduherminio) [inline src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md:125]:
```suggestion
## `SynchronizationContext` in xUnit Tests
```

File: src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md [hunk 2]
```diff
@@ -33,31 +33,109 @@ public void Advance(TimeSpan delta)
 public void SetLocalTimeZone(TimeZoneInfo localTimeZone)
 ```
 
-These can be used as follows:
+### ExpiryCache with TimeProvider
```
Comment (eduherminio) [inline src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md:36]:
```suggestion
### `ExpiryCache` with `TimeProvider`
```

**Review by eduherminio — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md [hunk 3]
```diff
@@ -33,31 +33,109 @@ public void Advance(TimeSpan delta)
...
+        var expirationTime = _timeProvider.GetUtcNow() + _expirationDuration;
+        var cacheItem = new CacheItem(value, expirationTime);
+
+        _cache[key] = cacheItem;
+    }
+
+    public bool TryGetValue(TKey key, out TValue value)
+    {
+        value = default;
+        if (_cache.TryGetValue(key, out var cacheItem))
```
Comment (RussKie) [inline src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md:68]: We don't use `var` unless the type is obvious.
Please refer to your [style guide](https://github.com/dotnet/extensions/blob/main/CONTRIBUTING.md#dos-and-donts).

**Review by RussKie — COMMENTED** [review]
👍

File: src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md (hunk 3 above)
Comment (amadeuszl) [reply src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md]: Fixed

File: src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md (hunk 2 above)
Comment (amadeuszl) [reply src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md]: Fixed

File: src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md (hunk 1 above)
Comment (amadeuszl) [reply src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md]: Fixed

**Review by RussKie — APPROVED** [review]

**Pull request merged**